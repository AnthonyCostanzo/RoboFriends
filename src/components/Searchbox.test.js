import {shallow,mount} from 'enzyme';
import React from 'react';
import SearchBox from './Searchbox';
import toJson from 'enzyme-to-json';

describe("SearchBox", () => {
    let wrapper;
    let mockSearchChange;
    beforeEach(() => {
      mockSearchChange = jest.fn();
      wrapper = shallow(<SearchBox searchChange={mockSearchChange} />);
    });
    it('should render',()=>
    {
        expect(shallow(<SearchBox searchChange = {mockSearchChange}/>).length).toEqual(1);
    })
    it("should match the snapshot", () => {
        expect(<SearchBox searchChange = {mockSearchChange}/>).toMatchSnapshot();
      });

      it('should pass a selected value to the onChange function', () => {
        const component = shallow(<SearchBox />);
        component.find('input').simulate('change', { target: {
          value: 'Change function' }
        });
    
        expect(toJson(component)).toMatchSnapshot();
      });
})
  

    //component.find('input').simulate('change');