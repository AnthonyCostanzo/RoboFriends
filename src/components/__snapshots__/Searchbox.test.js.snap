// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should match the snapshot 1`] = `
<SearchBox
  searchChange={[MockFunction]}
/>
`;

exports[`SearchBox should pass a selected value to the onChange function 1`] = `
<div
  className="pa2"
>
  <input
    aria-label="searchRobots"
    className="pa3 ba b--green bg-lighest-blue"
    placeholder="search robots"
    type="search"
  />
</div>
`;
